### PR TITLE
SM-92 admin grpc endpoints

### PR DESCRIPTION
**Pull request by lauradeng23**
# Proposed Changes

### CODE REVIEW

File: pkg/grpc/template_config.proto [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+}
+message UpdateTemplateResponse {
+  Template template = 1;
+
+}
+
+
+
+service TemplateConfigService {
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:57]: services look good. 
I think the CreateTemplate and UpdateTemplate could use the same UpdateTemplateRequest.
The response to those two could be a protobuf.empty - example [here](https://github.com/DIMO-Network/devices-api/blob/d4b8fbbc5e008ef378e51fd2effeecd68faa6060/pkg/grpc/user_devices.proto#L33)
, like i don't think there's much that needs to be returned.

File: pkg/grpc/template_config.proto [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  string id = 1;
+}
+
+message GetTemplateListRequest {
+  string powertrain = 1;
+  string protocol = 2;
+
+}
+message GetTemplateListResponse {
+  repeated Template templates = 1;
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:26]: I think the object we use for listing should be smaller and different than what we do for viewing/updating/creating. The reason is that if you picture the list of templates, these would just have the properties necessary for listing them in a table, but the properties you see when viewing the template (eg. clicking on the template) are going to be a lot more expanded.

File: pkg/grpc/template_config.proto (hunk 1 above)
Comment (lauradeng23) [reply pkg/grpc/template_config.proto:49]: updated to be just TemplateRequest

File: pkg/grpc/template_config.proto (hunk 2 above)
Comment (lauradeng23) [reply pkg/grpc/template_config.proto]: Added TemplateSummary to be redacted listing, also added name field

File: pkg/grpc/template_config.proto [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  string id  = 1;
+  string name = 2;
+  repeated string pids = 3;
+  string dbc = 4;
+  repeated string template_vehicles = 5;
+}
+message TemplateSummary {
+  string id = 1;
+  string name = 2;
+
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:20]: lets add the powertrain and protocol here too, how about also a hasDBC and PIDsCount? could be useful when we're looking at templates high level

File: pkg/grpc/template_config.proto [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc";
+
+package grpc;
+
+message Template {
+  string id  = 1;
+  string name = 2;
+  repeated string pids = 3;
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:13]:
```suggestion
  repeated PID pids = 3;
```

File: pkg/grpc/template_config.proto [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc";
+
+package grpc;
+
+message Template {
+  string id  = 1;
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:11]: do we need id if we have name? i think name is the pkey

File: pkg/grpc/template_config.proto [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  string id = 1;
+  string name = 2;
+
+}
+message PID {
+  string id = 1;
+}
+
+message GetTemplateListRequest {
+  string powertrain = 1;
```
Comment (Asfiroth) [inline pkg/grpc/template_config.proto:27]: are you planning to control if they're not sent in code? or this are gonna be optional

File: pkg/grpc/template_config.proto [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+package grpc;
+
+message Template {
+  string id  = 1;
+  string name = 2;
+  repeated string pids = 3;
+  string dbc = 4;
+  repeated string template_vehicles = 5;
+}
+message TemplateSummary {
```
Comment (Asfiroth) [inline pkg/grpc/template_config.proto:17]: we're missing some properties here, like version, parent, power train...

File: pkg/grpc/template_config.proto [hunk 8]
```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc";
+
+package grpc;
+
+message Template {
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:10]: missing a couple more properties that are in the DB table

File: pkg/grpc/template_config.proto (hunk 3 above)
Comment (lauradeng23) [reply pkg/grpc/template_config.proto:20]: added

**Comment by Asfiroth** [comment]
looks good for me now @JamesReate, @lauradeng23 please continue with rpc implementation

File: pkg/grpc/template_config.proto [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  string has_dbc = 5;
+  int32 pids_count = 6;
+
+}
+message PID {
+  string id = 1;
+}
+
+message GetTemplateListRequest {
+  string powertrain = 1;
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:38]:
```suggestion
  optional string powertrain = 1;
```

File: pkg/grpc/template_config.proto [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  string protocol = 2;
```
Comment (JamesReate) [inline pkg/grpc/template_config.proto:39]:
```suggestion
  optional string protocol = 2;
```

**Review by JamesReate — COMMENTED** [review]
just added a couple suggestions to make some properties optional, do let us know if you have questions for the next step of implementing the services.

File: pkg/grpc/template_config.proto (hunk 9 above)
Comment (lauradeng23) [reply pkg/grpc/template_config.proto]: done!

File: pkg/grpc/template_config.proto (hunk 10 above)
Comment (lauradeng23) [reply pkg/grpc/template_config.proto]: done!

**Pull request merged**